builder/vsphere/driver/vm.go

```diff
@@ -800,18 +800,24 @@ func (vm *VirtualMachine) addDevice(device types.BaseVirtualDevice) error {
 	return err
 }
 
-func (vm *VirtualMachine) AddConfigParams(params map[string]string) error {
+func (vm *VirtualMachine) AddConfigParams(params map[string]string, info *types.ToolsConfigInfo) error {
 	var confSpec types.VirtualMachineConfigSpec
 
-	var ov []types.BaseOptionValue
-	for k, v := range params {
-		o := types.OptionValue{
-			Key:   k,
-			Value: v,
+	if len(params) > 0 {
+		var ov []types.BaseOptionValue
+		for k, v := range params {
+			o := types.OptionValue{
+				Key:   k,
+				Value: v,
+			}
+			ov = append(ov, &o)
 		}
-		ov = append(ov, &o)
+		confSpec.ExtraConfig = ov
+	}
+
+	if info != nil {
```

Comment:
I think you don't even need to check the length of `params` and if `info` is nil here because these are the default values of confSpec.ExtraConfig and confSpec.Tools.

You could just check these before reconfiguring the VM to avoid doing it for nothing.
```
if len(confSpec.ExtraConfig) > 0 || confSpec.Tools != nil {
     task, err := vm.vm.Reconfigure(vm.driver.ctx, confSpec)
     ...
```
What do you think? 🤔